src/view/components/microbit/MicrobitSimulator.tsx

```diff
@@ -170,9 +175,56 @@ export class MicrobitSimulator extends React.Component<any, IState> {
         event.preventDefault();
         this.handleButtonClick(key, true);
     };
-
     protected onMouseLeave = (event: Event, key: string) => {
         event.preventDefault();
         console.log(`To implement onMouseLeave ${key}`);
     };
+    protected onKeyEvent(event: KeyboardEvent, active: boolean, key: string) {
+        event.stopPropagation();
+        if ([event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.ENTER)) {
+            this.handleButtonClick(key, active);
+            if (this.imageRef.current) {
+                if (key === BUTTONS_KEYS.BTN_A) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_A,
+                        active
+                    );
+                } else if (key === BUTTONS_KEYS.BTN_B) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_B,
+                        active
+                    );
+                } else if (key === BUTTONS_KEYS.BTN_AB) {
+                    this.imageRef.current.updateButtonAttributes(
+                        BUTTONS_KEYS.BTN_AB,
+                        active
+                    );
+                }
+            }
+        } else if (
+            [event.code, event.key].includes(CONSTANTS.KEYBOARD_KEYS.A)
+        ) {
+            this.handleButtonClick(BUTTONS_KEYS.BTN_A, active);
+            if (this.imageRef.current) {
+                this.imageRef.current.updateButtonAttributes(
+                    BUTTONS_KEYS.BTN_A,
+                    active
+                );

```

Comment:
You can use the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) in JavaScript to replace:
```
if (foo) {
  foo.bar();
}
```
with
`foo?.bar();`

Reply:
tried to implement it this way but kept on getting errors /:

Reply:
Yeah, it seems that we are running an older version of typescript. Up to you whether we should update it or not.

Reply:
I'll put it in the backlog. Thanks for the suggestion though 💃